routers/web/explore/repo.go

```diff
@@ -133,6 +135,7 @@ func RenderRepoSearch(ctx *context.Context, opts *RepoSearchOptions) {
 	ctx.Data["Total"] = count
 	ctx.Data["Repos"] = repos
 	ctx.Data["IsRepoIndexerEnabled"] = setting.Indexer.RepoIndexerEnabled
+	ctx.Data["Languages"] = enry.LanguagesColor
```

Comment:
You should only list the Languages that are used on this Gitea instance. There are a lot of small Instances which has only a few languages, so there is no need to have a full blown list. I can give you my Code if you want.

Reply:
Seems like you already have a better implementation than mine. If you would like to create a new PR, I would be quite pleased to close mine in favor of yours :)

Reply:
You can keep working on this, if you want.